draft-ietf-rats-corim.md

```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
-to a device, and a cryptographic key is the object. The predicate asserts that
-the identity is authenticated by the key. A common application for this triple
-is device identity.
+A Device Identity triple record relates one or more cryptographic keys to a device identity.
+The cryptographic keys are bound to or associated with a Target Environment that is within the device.
+The device identifier may be part of the Target Environment's `environment-map` or may be part of some other device identity credential, such as a certificate.
+The cryptographic keys are expected to be used to authenticate the device.
+
+Device Identity triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Device Identity triples.
+
+When Device Identity triples are contained in a CoRIM, it endorses that the keys were securely provisioned to the named Target Environment.
+Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as  `endorsed-triples`.
+
+Depending on key formatting, as defined by `$crypto-key-type-choice`, the Verifier may take different steps to locate and verify the key.
+
+If a key has usage restrictions that limits its use to device identity challenges, Verifiers SHOULD check for key use that violates usage restrictions.
+
+Offline verification of keys or verification of key use restrictions MAY produce Claims that are added to the ACS.
+Alternatively, Verifiers MAY report key verification results as part of an error reporting function.
 
 ~~~ cddl
 {::include cddl/identity-triple-record.cddl}
 ~~~
 
-#### Attestation Keys Triple {#sec-comid-triple-attest-key}
+#### Attest Key Triple {#sec-comid-triple-attest-key}
+
+An Attest Key triple record relates one or more cryptographic keys to an Attesting Environment.
+The cryptographic keys are wielded by an Attesting Environment that collects measurements from a Target Environment.
+The cryptographic keys sign Evidence.
+
+Attest Key triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Attest Key triples.
+
+When Attest Key triples are contained in a CoRIM, it endorses that the keys were securely provisioned to the named Attesting Environment.
+Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as  `endorsed-triples`.
+
+Depending on key formatting, as defined by `$crypto-key-type-choice`, the Verifier may take different steps to locate and verify the key.
+
+If a key has usage restrictions that limits its use to Evidence signing, Verifiers SHOULD check for key use that violates usage restrictions.
```

Comment:
To me this too is a MUST.

Reply:
See answer to the previous question about MUST